plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -119,12 +114,23 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
         }
     }
 
-    override fun embedFunction(symbol: FirFunctionSymbol<*>): FunctionEmbedding =
-        methods.getOrPut(symbol.embedName(this)) {
-            val signature = embedFullSignature(symbol)
-            val callable = processCallable(symbol, signature)
-            UserFunctionEmbedding(callable)
+    override fun embedFunction(symbol: FirFunctionSymbol<*>): FunctionEmbedding {
+        return when (val existing = methods[symbol.embedName(this)]) {
+            null, is PartiallySpecialKotlinFunction -> {
+                if (existing is PartiallySpecialKotlinFunction && existing.baseEmbedding != null)
+                    return existing
+                val signature = embedFullSignature(symbol)
+                val callable = processCallable(symbol, signature)
+                val userFunction = UserFunctionEmbedding(callable)
+                when {
+                    existing is PartiallySpecialKotlinFunction ->
+                        existing.also { it.initBaseEmbedding(userFunction) }
+                    else -> userFunction.also { methods[symbol.embedName(this)] = it }
+                }
```

Comment:
nit: it's a bit strange to have `when` here when you use `if` above for almost the same check.